gdal/swig/python/scripts/gdal2tiles.py

```diff
@@ -956,6 +1235,8 @@ def open_input(self):
         if not self.out_ds:
             self.out_ds = self.in_ds
 
+        self.out_ds.GetDriver().CreateCopy("gba.vrt", self.out_ds)
```

Comment:
Using a temporary file / unique filename would be better, otherwise there's a risk of collision if launching 2 gdal2tiles process from the same directory. For a single process approach, you could use a /vsimem/temporary_name.vrt to avoid hitting the disk. But if the file needs to be created once and shared by several processes (no idea here of what the code is doing !), then yes a on-disk file will be needed.